guides/source/7_1_release_notes.md

```diff
@@ -150,7 +150,51 @@ and reporting of the bulk enqueuing process.
 
 ### Composite primary keys
 
-TODO: Add description
+Applications can now create database tables with composite primary keys, and Rails will be able to derive these keys directly from the schema. This feature is particularly beneficial for many-to-many relationships and other complex data models where a single column is insufficient to uniquely identify a record.
+
+Active Record has been updated to fully support composite primary keys. All parts of the composite primary key will be included in the SQL generated by the `#reload`, `#update`, and `#delete methods`. Methods like `#first` and `#last` will use the full composite primary key in the `ORDER BY` statement.
+
+The `query_constraints`` macro can be used as a "virtual primary key" to achieve the same behaviors without modifying the database schema.
+
+Similarly, the `query_constraints:` option is accepted by associations to serve as a composite foreign key. This option configures the list of columns used for accessing the associated record.
+
+How to use:
+
+```ruby
+    create_table :travel_routes, primary_key: [:origin, :destination], force: true do |t|
+      t.string :origin
+      t.string :destination
+      t.text :description
+    end
+
+    create_table :travel_route_reviews, primary_key: [:origin, :destination], force: true do |t|
+      t.string :travel_route_origin
+      t.string :travel_route_destination
+      t.text :body
+      t.integer :author_id
+    end
+
+    class TravelRoute < ActiveRecord::Base; end
+
+    class TravelRouteReview < ActiveRecord::Base
+      belongs_to :travel_route, query_constraints: [:travel_route_origin, :travel_route_destination]
+    end
+
+    travel_route = TravelRoute.create(origin: "Ottawa", destination: "Montreal")
+    travel_route.id # => ["Ottawa", "Paris"]
+
+    travel_route.update(description: "Ottawa to Montreal route")
+    # UPDATE travel_routes SET description = 'Ottawa to Montreal route'
+    #  WHERE origin = 'Ottawa' AND destination = 'Montreal'
+
+    review = TravelRouteReview.new(author_id: 1, body: "I often take this route!")
+    review.travel_route = travel_route
+    review.travel_route_origin # => "Ottawa"
+    review.travel_route_destination # => "Montreal"
+
+    travel_route.destroy
+    # DELETE FROM travel_routes WHERE origin = 'Ottawa' and destination = 'Montreal'
+```
```

Comment:
Fix the indentation for the ruby block

Reply:
Thanks, we decided to remove this section from the notes but I removed the indentation from the new ones